### PR TITLE
Initial implementation of a X3D previewer with the X_ITE library

### PR DESCRIPTION
**Pull request by PaulBoon**
Adds a previewer for the x3d format (mime type: model/x3d+xml) using the X_ITE library. 
Using that library potentially many other formats can be rendered. 
Other libs I tried where x3dom and three.js, X_ITE was the simplest to use. 

Solves part of https://github.com/gdcc/dataverse-previewers/issues/55

Example (screendump): 

Screenshot from 2024-04-20 15-04-17

Tips for testing:
Use  the famous teapot model: 
https://www.web3d.org/x3d/content/examples/Basic/CAD/CadTeapot.x3d

### CODE REVIEW

**Comment by pdurbin** [comment]
@PaulBoon thanks for the pull request! I'm not sure if you say my post on the google group, but we are definitely in the market for a 3D viewer! Please see also:

- https://github.com/IQSS/dataverse.harvard.edu/issues/256
- https://groups.google.com/g/dataverse-community/c/D-vGQ5JP_7I/m/9uQcHTjVAAAJ

**Comment by qqmyers** [comment]
@PaulBoon - in terms of a PR - the v1.4 previewers dir is intended to be frozen (so that people deploying from the gdcc repo don't see any changes). If you make your changes in the betatest dir, they are available once merged, and will also get added to the 1.5 version when that comes out.

**Comment by landreev** [comment]
@PaulBoon very cool! In addition to ```"contentType":"model/x3d+xml"```, do you think this previewer could be used on all, or any of the other formats listed at https://create3000.github.io/x_ite/ under "Supported File Formats"?

**Comment by PaulBoon** [comment]
Thanks for those positive comments, I will update my 'Draft' to use a 1.5 directory and have a look at more types + test with larger models.

**Comment by PaulBoon** [comment]
OK, I fixed some things so now it should work with more mimetypes: 
- model/x3d+xml
- model/vrml
- model/gltf-binary
- model/obj
- model/stl
- model/x-ply

I moved the 'code' to the 'betatest' folder, but those curl calls reference `1.4`, so you would have to move the to the 1.4 folder when testing or change that curl.

File: previewers/betatest/X3DPreview.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <script type="text/javascript" src="js/retriever.js"></script>
+    <!-- Latest compiled and minified CSS -->
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css"
+        integrity="sha384-HSMxcRTRxnN+Bdg0JdbxYKrThecOKuH5zCYotlSAcp1+c8xmyTe9GYg1l9a69psu" crossorigin="anonymous">
+    <!-- Optional theme -->
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap-theme.min.css"
+        integrity="sha384-6pzBo3FDv/PJ8r2KRkGHifhEocL+1X2rVCTTkUfGk7/0pbek5mMa1upzvWbrUbOZ" crossorigin="anonymous">
+    <link type="text/css" rel="stylesheet" href="css/preview.css" />
+    <!-- X_ITE , latest via github -->
+    <script src="https://create3000.github.io/code/x_ite/latest/x_ite.min.js"></script>
```
Comment (qqmyers) [inline previewers/betatest/X3DPreview.html:20]: For production, they recommend using a cdn and picking a version. Right now the latest is at https://cdn.jsdelivr.net/npm/x_ite@9.5.2/dist/x_ite.min.js - does it make sense to switch this?

Comment (PaulBoon) [reply previewers/betatest/X3DPreview.html:20]: We could also download it and put it in the lib directory, not sure what best practice is. But for now a cdn seems fine.

File: previewers/betatest/X3DPreview.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+</head>
+
+<body class="container">
+    <main><img id='logo' alt='Site Logo'>
+        <h1 class="page-title x3dPreviewText">X3D Preview</h1>
```
Comment (qqmyers) [inline previewers/betatest/X3DPreview.html:25]: Should it just be 3D Preview now?

File: previewers/betatest/X3DPreview.html (hunk 1 above)
Comment (qqmyers) [reply previewers/betatest/X3DPreview.html:20]: Sounds good - I was more worried about /latest resulting in new code being loaded. I'm not sure we have a best practice overall - probably remote makes more sense for the version here, where people use it in hosted mode (getting the js from github.io versus a cdn isn't much difference), but people may want to use a local script if they're downloading everything here to run local.

**Review by qqmyers — APPROVED** [review]
With the minor changes requested and if you OK my doc edits, I'll go ahead and merge this. That will make it available in the betatest dir from github.io so it will be easier for people to try and do any QA.

**Comment by PaulBoon** [comment]
After downloading some example files for testing those mimetypes, I discovered that the preview does not always render them optimally. 
My expectations for a good previewer are that the initial rendering of a loaded file does try to let me see the model, by manipulating the viewport and lighting etc. for me. 
This does not always work, sometimes the model is far away, and I have to zoom in first. Other times the model is out of sight completely or the shading isn't helpful. 

Downloaded sample files from MeshLab at: https://www.meshlab.net/data/sample_meshes.zip
For glTF files I found this: 
https://github.com/KhronosGroup/glTF-Sample-Models/blob/main/2.0/README.md

**Comment by PaulBoon** [comment]
I just downloaded `MRIPAS_coll_170974_VulpesVulpes_2017_F_jaw_scan3D.stl ` from https://dataverse.openforestdata.pl/dataset.xhtml?persistentId=doi:10.48370/OFD/Y8TNUO. 
This DV has a 3D viewer. 
If I use the X3Dviewer I have to zoom out first, but then it renders fine.

**Comment by qqmyers** [comment]
OK - I'll merge to make it easier for people to test. If there's further work to set a useful initial default view or something, just create a new PR.  Thanks for the viewer - lots of people are interested!

**Pull request merged**